Fix various build issues and system incompatibilities

Patch by Gordon Bos

--- a/personal-setting-files.txt
+++ b/personal-setting-files.txt
@@ -1,22 +1,23 @@
 # network related
-/etc/network/interfaces
-/etc/network/firewall.conf
-/etc/network/easyfind.conf
+/etc/conf.d/net
+/etc/bubba/firewall.conf
+/etc/bubba/easyfind.conf
 /etc/dnsmasq.d/bubba.conf
-/etc/hostname
+/etc/conf.d/hostname
 /etc/hosts
 /etc/proftpd/proftpd.conf
 /etc/samba/smb.conf
 /etc/postfix/main.cf
-/etc/dhcp/dhclient.conf
+/etc/postfix/master.cf
+/etc/dhcpcd.conf
 /etc/cups/printers.conf
 /etc/resolv.conf
 /etc/fetchmailrc
 /etc/postfix/sasl_passwd
 /etc/postfix/sasl_passwd.db
-/etc/ftd/ftdconfig.ini
+/etc/bubba/ftdconfig.ini
 /etc/hostapd/hostapd.conf
-/etc/bubba-networkmanager.conf
+/etc/bubba/networkmanager.conf
 
 /etc/minidlna.conf
 
@@ -41,8 +42,8 @@
 /etc/apache2/cacert.pem
 /etc/apache2/privkey.pem
 
-# Squeezecenter
-/var/lib/squeezeboxserver/prefs
+# LogitechMediaServer
+/etc/squeezeboxserver
 
 # Bubba diverted configs
 /etc/bubba/*
--- a/web-admin/MANIFEST
+++ b/web-admin/MANIFEST
@@ -2,14 +2,11 @@
 MANIFEST
 lib/Bubba.pm
 bin/adminfunctions.php
-bin/updatebackend.pl
-bin/adminfunctions.php
 bin/disk.pl
 bin/firewall.pl
 bin/updater.pl
 bin/thumbnail.pl
 bin/backend.pl
-bin/print.pl
 bin/backup.pl
 bin/diskdaemon.pl
 bin/notify-dispatcher.pl
--- a/web-admin/Makefile.PL
+++ b/web-admin/Makefile.PL
@@ -8,7 +8,6 @@
 	'bin/backend.pl',
 	'bin/disk.pl',
 	'bin/firewall.pl',
-	'bin/easyfind.pl',
 	'bin/backup.pl',
 	'bin/diskdaemon.pl',
 	'bin/notify-dispatcher.pl',
--- a/web-admin/bin/adminfunctions.php
+++ b/web-admin/bin/adminfunctions.php
@@ -1302,7 +1302,7 @@
 
 function get_timezone_info() {
 	
-	$zoneinfo = '/usr/share/zoneinfo/right';
+	$zoneinfo = '/usr/share/zoneinfo';
 	if ($h_zonebase = opendir($zoneinfo)) {
 		
 		$zones = array();
--- a/web-admin/lib/Bubba.pm
+++ b/web-admin/lib/Bubba.pm
@@ -416,13 +416,9 @@
    my ($name)=@_;
    my $ret=0;
 
-   system("userdel $name");
+   system("smbpasswd -x $name &>/dev/null && userdel $name");
    $ret=$?;
    
-   if ($ret==0) {
-      system("smbpasswd -x $name");
-   }
-
    return $ret;
 }
 
@@ -826,13 +822,13 @@
 	my $pidfile;
 	
 	if ($service eq "fetchmail"){
-		$pidfile="/var/run/fetchmail/.fetchmail";
-	} elsif ($service eq "dnsmasq") {
-		$pidfile="/var/run/dnsmasq/dnsmasq.pid";
+		$pidfile="/var/run/fetchmail/fetchmail.pid";
 	} elsif ($service eq "avahi-daemon"){
 		$pidfile="/var/run/avahi-daemon/pid";
 	} elsif ($service eq "tor"){
 		$pidfile="/var/run/tor/tor.pid";
+	} elsif ($service eq "filetransferdaemon"){
+		$pidfile="/var/run/ftd.pid";
 	} else {
 		$pidfile = "/var/run/$service.pid";
 	}
@@ -1105,6 +1109,11 @@
 			print FILE "\tuser '".@{$line}[2]."' there with password '".@{$line}[3]."' is '".@{$line}[4]."' here ".@{$line}[5]." $keep\n";
    }
    close FILE;
+   # Gordon 2015-12-06: set correct owner and rights on fetchmailrc
+   chmod 0600, "/etc/fetchmailrc";
+   my $login, $passwd, $uid, $gid;
+   ($login,$pass,$uid,$gid) = getpwnam("fetchmail") or die "fetchmail not in passwd file";
+   chown $uid, $gid, "/etc/fetchmailrc";
 }
 
 
@@ -1442,8 +1443,8 @@
 
 	# services, boolean such if service enabled or not
 	my %services = map {
-		$_ => (defined bsd_glob "/etc/rc2.d/S??$_");
-	} qw(proftpd forked-daapd ntp filetransferdaemon cups postfix dovecot fetchmail minidlna dnsmasq logitechmediaserver hostapd netatalk ifup-br0 samba);
+		$_ => (defined -e "/etc/runlevels/default/$_");
+	} qw(proftpd forked-daapd ntpd filetransferdaemon cupsd postfix dovecot fetchmail minidlna dnsmasq logitechmediaserver hostapd netatalk net.br0 samba);
 
 	my $meta = {
 		version => $revision,
@@ -1659,8 +1660,8 @@
 			'--absolute-name'
 		);
 
-		system("/sbin/iptables-restore","/etc/network/firewall.conf");
-		system("/bin/cp","/etc/hostname","/proc/sys/kernel/hostname");
+		system("/sbin/iptables-restore","/etc/bubba/firewall.conf");
+		system("/usr/bin/rc-config","restart","hostname");
 
 		restart_network("eth0");
 		# hostapd needs to be started prior to restarting LANIF
@@ -1782,9 +1783,9 @@
 			$lan = _get_lanif;
 			restart_network($lan);
 
-			system("/sbin/iptables-restore < /etc/network/firewall.conf");
+			system("/sbin/iptables-restore < /etc/bubba/firewall.conf");
 
-			system("echo `cat /etc/hostname` > /proc/sys/kernel/hostname");
+			system("/usr/bin/rc-config","restart","hostname");
 
 			if($lines=~/proftpd/){
 				start_service("proftpd");
@@ -1836,12 +1837,12 @@
 				remove_service("fetchmail");
 			}
 
-			if($lines=~/cups/){
-				start_service("cups");
-				reload_service("cups");
+			if($lines=~/cupsd/){
+				start_service("cupsd");
+				reload_service("cupsd");
 			}else{
-				stop_service("cups");
-				remove_service("cups");
+				stop_service("cupsd");
+				remove_service("cupsd");
 			}
 
 			if($lines=~/dnsmasq/){
